packages/flutter_tts/tizen/src/text_to_speech.h

```diff
@@ -8,33 +8,28 @@
 #include <tts.h>
 
 #include <functional>
+#include <optional>
 #include <string>
 #include <vector>
 
+enum class TtsState { kCreated, kReady, kPlaying, kPaused };
+
 using OnStateChangedCallback =
     std::function<void(tts_state_e previous, tts_state_e current)>;
 using OnUtteranceCompletedCallback = std::function<void(int utt_id)>;
 using OnErrorCallback = std::function<void(int utt_id, tts_error_e reason)>;
 
 class TextToSpeech {
  public:
-  TextToSpeech() {
-    // TODO : Handle initialization failure cases
-    // Rarely, initializing TextToSpeech can fail. IMO, in this case,
-    // we should throw an exception which means that the TextToSpeech instance
-    // creation failed. In addition, we should consider catching the exception
-    // and propagating it to the flutter side. however, I think this is optional
-    // because flutter side is not expecting any errors.
-    Init();
-    RegisterTtsCallback();
-    Prepare();
-  }
+  TextToSpeech() = default;
 
   ~TextToSpeech() {
     UnregisterTtsCallback();
-    Deinit();
+    TtsDestroy();
   }
 
+  bool Initialize();
+
   void SetOnStateChanagedCallback(OnStateChangedCallback callback) {
```

Comment:
https://github.com/flutter-tizen/plugins/pull/387#discussion_r882489624
```suggestion
  void SetStateChanagedCallback(StateChangedCallback callback) {
```

Reply:
done